[MRG+1]: fix event sample indices read from eeglab files

mmagnuski opened the pull request:
fix event latency from matlab 1-based to python 0-based.
Fixes #4944.

jasmainak commented on mne/io/eeglab/tests/test_eeglab.py:
```diff
@@ -121,8 +121,11 @@ def test_io_set():
+                               event_id=event_id, preload=True)
+
+    # test that sample indices are read python-wise (zero-based)
+    assert mne.find_events(test_raw)[0, 0] == eeg.event[0].latency - 1
```
can you add a test to check they are all positive?

mmagnuski replied:
sure, or do you prefer to throw an error **during reading** if some come out negative?

jasmainak replied:
hmm ... I would think we have a bug if something comes out negative _during reading_

mmagnuski replied:
Right, I'll add a ValueError to eeglab reader if latency comes out negative. I'm wondering if it is possible though due to how eeglab handles downsampling data (latency is then represented as float, so `int(0.7)` would be 0 and then -1 after index correction). But that is an issue for a next PR (for example rounding float latencies).

agramfort commented on mne/io/eeglab/eeglab.py:
```diff
@@ -717,15 +717,15 @@ def _read_annotations_eeglab(eeg):
         description = [str(eeg.event.type)]
-        onset = [eeg.event.latency]
+        onset = [eeg.event.latency - 1]
         duration = getattr(eeg.event, 'duration', np.zeros(1))
```
as you can see there is something broken here as onset in Annotations should be in seconds but I will fix it in https://github.com/mne-tools/mne-python/pull/4936

mmagnuski replied:
Ah, right. Ok, so I'll leave it the way it is now here.

agramfort commented:
can you set the PR to MRG and update what's new? thanks

jona-sassenhagen commented on mne/io/eeglab/eeglab.py:
```diff
@@ -717,15 +717,15 @@ def _read_annotations_eeglab(eeg):
     elif isinstance(eeg.event, np.ndarray):
         description = [str(event.type) for event in eeg.event]
-        onset = [event.latency for event in eeg.event]
+        onset = [event.latency - 1 for event in eeg.event]
```
Maybe add a comment here, just to be sure?

mmagnuski replied:
see a comment by @agramfort, few lines below, I'll leave it as it is for now.

jona-sassenhagen commented:
Looks good to me.

mmagnuski commented:
Last change was only whats_new so the CIs should come back green.

mmagnuski commented:
@agramfort - one last thing, @jasmainak asked me to throw an error when any time sample becomes negative during reading - should I do it here or do you prefer to deal with it in your PR?

mmagnuski commented:
Ok, I've added that ValueError - although I am wondering if it shouldn't be just a warning (for example we only give warnings about boundary events). Throwing error in the function that reads events would make it harder for someone to investigate what is wrong with the event latencies (but we can advise using sicpy.io.loadmat then) unless one has access to matlab on the same computer.
I've also added a test, but it will probably go back red as I have problems running tests locally :(

jasmainak commented:
They can just use postmortem debugging to investigate. And if they give us a reproducible script, it's not too hard :)

jasmainak commented:
@mmagnuski +1 for merge from me after circle is happy :)

mmagnuski commented:
@jasmainak Ok, I finally corrected the test errors and one pep8 violation. My bold prediction is that CIs will be green finally. @agramfort @jona-sassenhagen I'm ready for last checks if you want to take another look.

agramfort commented:
+1 for MRG

mmagnuski commented:
🥗 📗 all green 💚 🍏

larsoner commented:
Thanks @mmagnuski

mmagnuski commented:
Thanks! 🚀

jasmainak commented:
Thanks @mmagnuski :-)
